### PR TITLE
run DST via DSTEd on linux

### PR DESCRIPTION
**Pull request by ThemInspectors**
- it can be revised,
- OSX not included.

### CODE REVIEW

File: Source/Classes/Core.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
-﻿(function Core() {
+﻿const os2 = require('os'); // use a custom one, can be removed/replaced
```
Comment (Bizarrus) [inline Source/Classes/Core.js]: Already defined:

 > https://github.com/ThemInspectors/DSTEd/blob/fa263e2622913c1b4dd0d90efc453911faa96be0/Source/Classes/Core.js#L14

File: Source/Classes/Core.js [hunk 2]
```diff
@@ -130,15 +132,15 @@
 								Logger.info(data);
 							});**/
 							
-							if (os.platform() === 'win32') {
+							if (os2.platform() === 'win32') {
```
Comment (Bizarrus) [inline Source/Classes/Core.js]: Replace `os2` with `OS`.

File: Source/Classes/Core.js [hunk 3]
```diff
@@ -130,15 +132,15 @@
...
 								const ls = Process.spawn('D:\\Software\\Steam\\Steam.exe', [ '-applaunch', 322330, '-window' ], {
 									cwd: 'D:\\Software\\Steam\\',
 									detached: false
 								});
 							}
-							if (os.platform() === 'linux') {
-								const ls = Process.spawn(os.homedir() + '/.steam/steam/steamapps/common/Don''t Starve Together/bin/dontstarve', [ ], {
-									cwd: os.homedir() + '/.steam/steam/steamapps/common/Don''t Starve Together/bin/dontstarve',
+							if (os2.platform() === 'linux') {
```
Comment (Bizarrus) [inline Source/Classes/Core.js]: Replace `os2` with `OS`.

File: Source/Classes/Core.js [hunk 4]
```diff
@@ -130,15 +132,15 @@
...
+								const ls = Process.spawn('steam', [ 'steam://rungameid/322330'], {
+									cwd: os2.homedir(),
```
Comment (Bizarrus) [inline Source/Classes/Core.js]: Replace `os2` with `OS`.

File: Source/Classes/Core.js [hunk 5]
```diff
@@ -130,15 +132,15 @@
...
 								const ls = Process.spawn('D:\\Software\\Steam\\Steam.exe', [ '-applaunch', 322330, '-window' ], {
 									cwd: 'D:\\Software\\Steam\\',
 									detached: false
 								});
 							}
-							if (os.platform() === 'linux') {
-								const ls = Process.spawn(os.homedir() + '/.steam/steam/steamapps/common/Don''t Starve Together/bin/dontstarve', [ ], {
-									cwd: os.homedir() + '/.steam/steam/steamapps/common/Don''t Starve Together/bin/dontstarve',
+							if (os2.platform() === 'linux') {
+								const ls = Process.spawn('steam', [ 'steam://rungameid/322330'], {
```
Comment (Bizarrus) [inline Source/Classes/Core.js]: Wrong call; We wan't to debug the output of the executable from `Steam`. The `steam://`-Protocol won't start the game directly, Steam run's the game from the background-process.

**Review by Bizarrus — CHANGES_REQUESTED** [review]
Please see the Notes for approval.

File: Source/Classes/Core.js (hunk 5 above)
Comment (ThemInspectors) [inline Source/Classes/Core.js]: Don't see the need? I think I've already done that. The first commit was just testing to try and get things ready, although the 2nd commit was making sure things are right, unless I'm missing something I think you should be checking and edtting the final commit

**Comment by ThemInspectors** [comment]
Don't see the need? I think I've already done that. The first commit "Update Core.js" was just testing to try and get things ready, although the 2nd commit "Let DST run on Linux" was making sure things are right, unless I'm missing something I think you should be checking and editing the final commit. Although, yes, the "os2" is redundant and could be in the function.

File: Source/Classes/Core.js (hunk 4 above)
Comment (ThemInspectors) [reply Source/Classes/Core.js]: done.

File: Source/Classes/Core.js (hunk 5 above)
Comment (ThemInspectors) [reply Source/Classes/Core.js]: don't really get what you're asking? Would detaching it work?

File: Source/Classes/Core.js (hunk 1 above)
Comment (ThemInspectors) [reply Source/Classes/Core.js]: done.

File: Source/Classes/Core.js (hunk 2 above)
Comment (ThemInspectors) [reply Source/Classes/Core.js]: done.

**Pull request merged**